api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -196,5 +197,174 @@ public IActionResult Delete(string path, [FromRoute] string? location = "capture
         System.IO.File.Delete(safePath);
         return new JsonResult(info);
     }
+
+    /// <summary>
+    /// Make a clip from the target file identified in the clip parameter.
+    /// </summary>
+    /// <param name="fileName"></param>
+    /// <param name="directory"></param>
+    /// <param name="start"></param>
+    /// <param name="end"></param>
+    /// <param name="clipNbr"></param>
+    /// <returns></returns>
+    [HttpGet("clip")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(FolderModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Storage" })]
+    public IActionResult CreateClip(string fileName, string directory, int start, int end, int clipNbr, string target)
+    {
+        var path = _config.CapturePath + directory;
+        var process = GetClipProcess(fileName, path, start, end, clipNbr, target);
+
+        process.Start();
+        process.WaitForExit();
+
+        var listing = new FolderModel(_config.CapturePath, directory);
+        return new JsonResult(listing);
+    }
+
+    /// <summary>
+    /// Make a clip from the target file identified in the clip parameter.
+    /// </summary>
+    /// <param name="fileName"></param>
+    /// <param name="directory"></param>
+    /// <returns></returns>
+    [HttpPut("join")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(FolderModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Storage" })]
+    public IActionResult JoinClips(string fileName, string directory, string target)
+    {
+        var safePath = System.IO.Path.Combine(_config.GetRootPath("storage"), directory.MakeRelativePath());
+        var format = Path.GetExtension(fileName).Replace(".", "");
+
+        var muxFile = GenerateMuxfile(fileName, safePath, format, target);
+        var process = GetJoinProcess(directory, fileName, safePath, muxFile, format, target);
+
+        process.Start();
+        process.WaitForExit();
+
+        System.IO.File.Delete(muxFile);
+
+        var listing = new FolderModel(_config.CapturePath, directory);
+        return new JsonResult(listing);
+    }
+    #endregion
+
+    #region Support Routines
```

Comment:
Nervous about adding all this logic to the StorageController.  An API controller should focus on a very narrow purpose, listen, respond.  The majority of logic should be moved to the DAL.

Reply:
Fixed.